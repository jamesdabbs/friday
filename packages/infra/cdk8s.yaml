language: typescript
app: node main.js
imports:
  - k8s
  - https://strimzi.io/install/latest?namespace=kafka
