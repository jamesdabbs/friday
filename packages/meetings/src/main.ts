import { AppModule } from './app.module';
import { bootstrap } from '@jamesdabbs/nest-platform'

bootstrap(AppModule, { hot: (module as any).hot })
