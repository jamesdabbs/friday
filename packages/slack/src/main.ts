import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';
import { Transport, MicroserviceOptions } from '@nestjs/microservices';

async function bootstrap() {
  const app = await NestFactory.createMicroservice<MicroserviceOptions>(AppModule, {
    transport: Transport.KAFKA,
    options: {
      client: {
        brokers: ['kafka-cluster-kafka-bootstrap.kafka.svc.cluster.local:9092'],
      },
      consumer: {
        groupId: 'slack'
      }
    }
  });

  app.listen(() => {
    console.log('ðŸ¤– Slack is listening')
  })
}
bootstrap();
