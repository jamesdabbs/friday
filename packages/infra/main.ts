import { Construct } from "constructs";
import { App, Chart, ChartProps } from "cdk8s";
import { KafkaChart } from "./src/kafka.chart";
import { SlackChart } from "./src/slack.chart";
import { MeetingsChart } from "./src/meetings.chart";
import { ScheduleChart } from "./src/schedule.chart";
import { MinutesChart } from "./src/minutes.chart";

export class MyChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = {}) {
    super(scope, id, props);

    const manageKafka = false;
    if (manageKafka) {
      new KafkaChart(this, "kafka.chart");
    }

    new MeetingsChart(this, "meetings.chart", {
      // schedule: "*/1 * * * *"
    });
    new MinutesChart(this, "minutes.chart");
    new ScheduleChart(this, "schedule.chart");
    new SlackChart(this, "slack.chart");
  }
}

const app = new App();
new MyChart(app, "infra");
app.synth();
