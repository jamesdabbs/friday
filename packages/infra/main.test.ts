import { FridayChart } from './main';
import { Testing } from 'cdk8s';

describe('Placeholder', () => {
  test('with defaults', () => {
    const app = Testing.app();
    const chart = new FridayChart(app, 'test-chart');
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
