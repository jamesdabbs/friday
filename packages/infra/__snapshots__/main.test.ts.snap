// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder with defaults 1`] = `
Array [
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "meetings-db",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 5432,
        },
      ],
      "selector": Object {
        "cdk8s.statefulset": "test-chart-meetings.chart-meetings.db.sts-c8c94a82",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "StatefulSet",
    "metadata": Object {
      "name": "meetings-db",
    },
    "spec": Object {
      "podManagementPolicy": "OrderedReady",
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.statefulset": "test-chart-meetings.chart-meetings.db.sts-c8c94a82",
        },
      },
      "serviceName": "meetings-db",
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.statefulset": "test-chart-meetings.chart-meetings.db.sts-c8c94a82",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "POSTGRES_PASSWORD",
                  "value": "hunter2",
                },
              ],
              "image": "postgres",
              "imagePullPolicy": "IfNotPresent",
              "name": "postgres",
              "ports": Array [
                Object {
                  "containerPort": 5432,
                },
              ],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "meetings",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "meetings.chart-meetings.chart.nest-meetings.deployment-c83e5c65",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "meetings.chart-meetings.chart.nest-meetings.deployment-c83e5c65",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "DATABASE_URL",
                  "value": "postgres://postgres:hunter2@meetings-db:5432",
                },
              ],
              "image": "meetings",
              "imagePullPolicy": "IfNotPresent",
              "livenessProbe": Object {
                "failureThreshold": 3,
                "httpGet": Object {
                  "path": "/health",
                  "port": 3000,
                },
              },
              "name": "meetings",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "meetings",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "cdk8s.deployment": "meetings.chart-meetings.chart.nest-meetings.deployment-c83e5c65",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "name": "meetings.ingress.local",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "meetings.localhost",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "meetings",
                  "servicePort": 80,
                },
              },
            ],
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "batch/v1beta1",
    "kind": "CronJob",
    "metadata": Object {
      "name": "meetings-daily-reminder",
    },
    "spec": Object {
      "jobTemplate": Object {
        "spec": Object {
          "template": Object {
            "spec": Object {
              "containers": Array [
                Object {
                  "command": Array [
                    "curl",
                    "-XPOST",
                    "meetings/sendDailyReminders",
                  ],
                  "image": "curlimages/curl",
                  "imagePullPolicy": "IfNotPresent",
                  "name": "daily-reminder",
                },
              ],
              "restartPolicy": "Never",
            },
          },
          "ttlSecondsAfterFinished": 60,
        },
      },
      "schedule": "0 9 * * *",
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "minutes",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "te-minutes.chart-minutes.chart.nest-minutes.deployment-c87169d8",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "te-minutes.chart-minutes.chart.nest-minutes.deployment-c87169d8",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "CONFLUENCE_EMAIL",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "CONFLUENCE_EMAIL",
                      "name": "minutes-secret",
                    },
                  },
                },
                Object {
                  "name": "CONFLUENCE_API_TOKEN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "CONFLUENCE_API_TOKEN",
                      "name": "minutes-secret",
                    },
                  },
                },
                Object {
                  "name": "CONFLUENCE_DOMAIN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "CONFLUENCE_DOMAIN",
                      "name": "minutes-secret",
                    },
                  },
                },
              ],
              "image": "minutes",
              "imagePullPolicy": "IfNotPresent",
              "livenessProbe": Object {
                "failureThreshold": 3,
                "httpGet": Object {
                  "path": "/health",
                  "port": 3000,
                },
              },
              "name": "minutes",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "minutes",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "cdk8s.deployment": "te-minutes.chart-minutes.chart.nest-minutes.deployment-c87169d8",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "name": "minutes.ingress.local",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "minutes.localhost",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "minutes",
                  "servicePort": 80,
                },
              },
            ],
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "schedule",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "test-chart-schedule.chart-schedule.deployment-c8c56cb1",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "test-chart-schedule.chart-schedule.deployment-c8c56cb1",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "GOOGLE_CLIENT_ID",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "GOOGLE_CLIENT_ID",
                      "name": "schedule-secret",
                    },
                  },
                },
                Object {
                  "name": "GOOGLE_CLIENT_SECRET",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "GOOGLE_CLIENT_SECRET",
                      "name": "schedule-secret",
                    },
                  },
                },
                Object {
                  "name": "GOOGLE_PERSONAL_REFRESH_TOKEN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "GOOGLE_PERSONAL_REFRESH_TOKEN",
                      "name": "schedule-secret",
                    },
                  },
                },
                Object {
                  "name": "GOOGLE_CLIENT_REDIRECT_URI",
                  "value": "http://localhost:3001/auth/google/callback",
                },
              ],
              "image": "schedule",
              "imagePullPolicy": "IfNotPresent",
              "name": "schedule",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "schedule",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "cdk8s.deployment": "test-chart-schedule.chart-schedule.deployment-c8c56cb1",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "name": "schedule.ingress.local",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "schedule.localhost",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "schedule",
                  "servicePort": 80,
                },
              },
            ],
          },
        },
      ],
    },
  },
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "slack",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "test-chart-slack.chart-slack.deployment-c880188c",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "test-chart-slack.chart-slack.deployment-c880188c",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "SLACK_SIGNING_SECRET",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "SLACK_SIGNING_SECRET",
                      "name": "slack-secret",
                    },
                  },
                },
                Object {
                  "name": "SLACK_BOT_USER_TOKEN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "SLACK_BOT_USER_TOKEN",
                      "name": "slack-secret",
                    },
                  },
                },
              ],
              "image": "slack",
              "imagePullPolicy": "IfNotPresent",
              "name": "slack",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
]
`;
